scripts/gen_kobject_list.py

```diff
@@ -269,10 +276,14 @@ def main():
     parse_args()
 
     if args.gperf_output:
+        assert args.kernel, "--kernel ELF required for --gperf-output"
```

Comment:
Maybe

```python
if not args.kernel:
    sys.exit("--kernel must be passed if --gperf-output is")
```

would be a bit more purist.

It's more of a usage error than a runtime error.

Reply:
It is a usage error but... a pretty stupid, "trial-and-error" one so I didn't want to spend too many lines of code on it.